src/pybind/mgr/cephadm/tests/test_scheduling.py

```diff
@@ -379,19 +379,22 @@ class NodeAssignmentTest(NamedTuple):
     placement: PlacementSpec
     hosts: List[str]
     daemons: List[DaemonDescription]
+    rank_map: Optional[Dict[int, Dict[int, Optional[str]]]]
+    post_rank_map: Optional[Dict[int, Dict[int, Optional[str]]]]
     expected: List[str]
     expected_add: List[str]
     expected_remove: List[DaemonDescription]
 
 
-@pytest.mark.parametrize("service_type,placement,hosts,daemons,expected,expected_add,expected_remove",
+@pytest.mark.parametrize("service_type,placement,hosts,daemons,rank_map,post_rank_map,expected,expected_add,expected_remove",
```

Comment:
Bad feeling about that. This is going to make the test matrix explode.

Reply:
for the non-ranked test cases, these are just None, None.  I thought about making a separate test but that would just duplicate the test code... or put it in a helper that just passed None, None for the older test cases.

Reply:
I'm concerned that we essentially have to test

(different number of hosts) * (different existing placement on those hosts) * (different networks) * (different filter_new_host functions) * (allow_colo = yes and allow_colo = no) * (primary_daemon_type) * (per_host_daemon_type) * (different values for the rank_map). 

And that scares me.